README.md

```diff
@@ -0,0 +1,203 @@
+
+## Pipe 1: Find unreferenced statements and URLs for a given list of Wikidata items<a name="pipe-1"></a>
+
+This step will take in the following inputs, and will output the data expected in [Pipe 2](#pipe-2).
+
+* `<varname>`: List of strings representing white listed Wikidata external id properties, same as output of [SS1](#ss-1).
+
+* `<varname>`: List of Wikidata item serializations, as described in https://www.mediawiki.org/wiki/Wikibase/DataModel/JSON
+
+## Pipe 2: Scrape given URLs for potential matches between  unreferenced statements and structured data<a name="pipe-2"></a>
+
+This step will take in the following inputs and will output the data expected in [Pipe 3](#pipe-3).
+
+* `<varname>`:  List of objects to represent Wikidata Items with the following keys:
+
+  * `itemId`: The Wikidata id of the item being examined
+
+  * `resourceUrls`: List of objects describing external resources to scrape, and reference meta data to be added to references.
+
+    * `url`: String representing the url to scrape
+    * `referenceMetadata`: Data to append to the potential references matched at the end of this stage with optional reference keys. 
+
+    ***Note***: An example for this property can be found in the output provided by [SS4](#ss-4).
+    
+  * `statements`: List of unreferenced statements in the following structure:
+    
+    * `pid`: The property id of the statement
+    * `dataValue`: The data value of the statement / Item id of the Wikidata item value.
+  
+    Example:
+  
+      ```json
+    [
+        {
+            "pid": "P321", 
+            "dataValue": "2020-04-01",
+        },
+        {
+            "pid": "P777",
+            "dataValue": "Q666"
+        }
+        // ...
+    ]
+      ```
+
+## Pipe 3: Filter extracted values by plain text matching<a name="pipe-3"></a>
+
+This step will take the output of Pipe 2 as it's primary input. It will also take the output of SS2 as an input.
+
+One row of this file relates to one unreferenced statement. It contains the itemId the statement was on as well as all the extracted structured data that relates to this statement and referenceMetadata referring to where that data came from.
+
+The format of the template of one row of Pipe 2 is as follows:
+```json
+{
+  "statement": {statementBlob},
+  "itemId": [itemId],
+  "references": [ {
+    referenceMetadata: referenceBlob,
+    extractedData: [ [value] ]
+    } ]
+}
+```
+
+### referenceBlob format
+```json
+{
+  *[statedInPropId]: [externalIdItem],
+  *[externalIdProp]: [externalIdVal],
+  *[referenceUrl]: [String],
+  [dateRetrieved]: [String]
+}
+```
+
+### statementBlob format
+```json
+{
+  "propertyId": [String],
+  "datatype": [String],
+  "value": {valueBlob}
+}
+```
+
+### valueBlob
+See https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/Wikibase/+/master/docs/topics/json.md#Snaks-json_snaks for the datavalue -> value key.
```

Comment:
Bare links are not great in README. I would hide under something like "related documentation"

Reply:
Sure thing.

Reply:
Both links to json docs now have a pretty name